webapp/src/utils.ts

```diff
@@ -121,9 +121,8 @@ class Utils {
 
     // Date and Time
 
-    static displayDate(date: Date, intl: IntlShape, dateFormat: string): string {
-        const text = intl.formatDate(date, {year: 'numeric', month: 'short', day: '2-digit', format: dateFormat})
-
+    static displayDate(date: Date, intl: IntlShape): string {
+        const text = intl.formatDate(date, {year: 'numeric', month: 'numeric', day: '2-digit'})
```

Comment:
The previous code that added `format: dateFormat` caused a javascript error here. This function will return the correct date format based on locale. This function currently is only being used for read-only views of date properties and displays the month as MMM (ie. 'Jul')

The Day Picker Input handled formatting the date as either 'MM/DD/YYYY' or 'DD/MM/YYYY' based on user setting. This function now provides all date formatting for the EditableDayPickers. It was changed to match the current date picker format and will return the correct format based on Locale. So the user setting is no longer necessary. - https://github.com/mattermost/focalboard/pull/666 

We should eventually allow user to pick date format, but not for locale, but for short vs 2-digit month, etc.

Reply:
@sbishel is there any reason to ignore user preferred format? If user has set DD/MM/YYYY then we should show in that format right?

Reply:
Actually, it looks like this has changed with this [PR](https://github.com/mattermost/focalboard/pull/791) I will be updating to latest main and will rework based on the latest.

Reply:
The preferred user format was removed with this [PR](https://github.com/mattermost/focalboard/pull/790). This PR is supporting dates in the a similar manner. However, it parses and displays dates for editing using the locale "L" Date format. Typically "MM/DD/YYYY" or "DD/MM/YYYY" depending on local. 
When empty the placeholder will show the parsable format.